scripts/cleanall.sh

```diff
@@ -11,6 +11,9 @@ rm -rf release
 rm -f bin/flatcc*
 rm -f bin/bfbs2json*
 rm -f lib/libflatcc*
+rm -f bin/*test*
+rm -f bin/flatc_compat*
+rm -f bin/monster*
 if [ -d bin ]; then
```

Comment:
Why do these tests end up here in the first place? That was never the intention with the bin dir.

Reply:
When building shared libraries on Windows, the tests failed because they could not find `libflatccruntime.dll`.
Maybe this can be revisited by setting the PATH environment variable to `$prefix/lib`.

Reply:
Wouldn't it make more sense to add $prefix/lib to PATH when it comes to DLL search paths?

Reply:
Yes, I'm looking into that atm.

I tried that first yesterday, but couldn't get it to work with [`msys2/setup-msys2@v2`](https://github.com/msys2/setup-msys2).
But I'm not using it anymore, so it's fine.

Reply:
But that didn't answer why tests a stored in the bin directory.

Reply:
Another fine approach would be to use the `ENVIRONMENT` property on targets.
See https://cmake.org/cmake/help/latest/prop_test/ENVIRONMENT.html

This can be done very easily with the postponed test patches (=doing diff analysis).

Reply:
> But that didn't answer why tests a stored in the bin directory.

It just avoided setting PATH.

Reply:
As I recall, all executables were built somewhere in the build tree, and only flatcc had a specific output target to bin.
Now it seems that all executables default to bin?

Reply:
Yes, for mingw on WIndows.
I'm testing whether this can be reverted.

Reply:
By the way, building these inside the source tree causes problems when configure cmake in 2 folders, as ci does for cross building.
Both might build `libflatcc.a` and `libflatccrt.a` in the `lib` subfolder.

Imho it's better to build these in a build tree subfolder.

Reply:
The logical thing would be to define a separate bin, lib location for cross target. They definitely should not be the same location.

Reply:
That's a mess.

You can also have 2 native cmake build trees in parallel with different configuration paramenters.
Those would then conflict.
Keeping the binaries in the build tree is the logical thing.
You have the install target to export everything.